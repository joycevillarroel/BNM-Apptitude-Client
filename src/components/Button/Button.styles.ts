import styled from 'styled-components'

interface IButtonContainerProps {
  selected: boolean
}

export const ButtonContainer = styled.button<IButtonContainerProps>`
  background-color: ${({ theme, selected }) =>
    selected ? theme.primary : theme.gray200};
  min-width: 150px;
  border: none;
  height: 45px;
  border-radius: 4px 4px 0px 0px;
`
