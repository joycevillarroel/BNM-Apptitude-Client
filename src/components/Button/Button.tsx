import { ButtonContainer } from './Button.styles'

interface IButton {
  selected?: boolean
  onClick: () => void
  children: string
}

export function Button({ selected = true, children, onClick }: IButton) {
  return (
    <ButtonContainer selected={selected} onClick={onClick}>
      {children}
    </ButtonContainer>
  )
}
