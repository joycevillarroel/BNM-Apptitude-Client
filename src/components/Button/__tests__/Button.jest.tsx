import { fireEvent, render } from "@testing-library/react"
import { Button } from '../Button'

describe('Button', () => {
  const mockFn = jest.fn()
  test('should render', () => {
    const { getByText } = render(<Button onClick={mockFn}>test</Button>)

    expect(getByText('test')).toBeTruthy()
    expect(getByText('test')).toMatchSnapshot()
  })

  test('should call onClick function', () => {
    const { getByText } = render(<Button onClick={mockFn}>test</Button>)
    const button = getByText('test')
    fireEvent.click(button)
    expect(mockFn).toBeCalledTimes(1)
  })
})
