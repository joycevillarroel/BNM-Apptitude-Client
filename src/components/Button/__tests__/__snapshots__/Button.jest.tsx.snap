// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button should render 1`] = `
<button
  class="sc-beqWaB hafbjw"
>
  test
</button>
`;
